src/open_inwoner/openproducten/tests/helpers.py

```diff
@@ -0,0 +1,147 @@
+from datetime import date
+from uuid import uuid4
+
+from django.core.files import File as DjangoFile
+from django.core.files.temp import NamedTemporaryFile
+
+from filer.models.filemodels import File as FilerFile
+
+from open_inwoner.openproducten import models as op_models
+from open_inwoner.openproducten.api_models import (
+    Category,
+    Condition,
+    File,
+    Link,
+    Price,
+    PriceOption,
+    ProductType,
+    Question,
+    Tag,
+    TagType,
+)
+from open_inwoner.pdc import models as pdc_models
+
+
+def create_file_object(content):
+    temp_file = NamedTemporaryFile(delete=True)
+    temp_file.write(content)
+    return DjangoFile(temp_file)
```

Comment:
Goed om een context manager te gebruiken voor temp files, zodat Python ze netjes kan opruimen na afloop:

```suggestion
    with NamedTemporaryFile(delete=True) as temp_file:
        temp_file.write(content)
        return DjangoFile(temp_file)
```

Reply:
Met een context manager ontstaat een error als een bestand later wordt aangemaakt in de media_root.

Om alle bestanden te verwijderen is nu de MEDIA_ROOT overschreven zodat het na de testen kan worden verwijderd.

in `TestOpenProductenImporter` wordt `create_temp_file` direct aangeroepen waardoor de bestanden niet uit `/tmp/` worden verwijderd. daarom wordt dit nu gedaan in de teardown.

in `TestProductTypeImporter` wordt `create_file_instance` aangeroepen waardoor ze wel worden verwijderd uit `/tmp/`.

Daarnaast hebben een aantal testen in `TestProductTypeImporter` een integrity error in de post_teardown, waarschijnlijk omdat het bestand al wel weg is. Dit is nu opgelost door alle ProductFiles alvast te verwijderen in de tearDown, geen idee waarom dit wel werkt.

Reply:
Zou je kunnen proberen om, ipv een `DjangoFile`, [een `ContentFile` te gebruiken](https://docs.djangoproject.com/en/5.1/ref/files/file/#the-contentfile-class)? Dan zit alles in-memory en heb je het hele gedonder niet van die tempfiles.

Reply:
(De constructie waarbij je `MEDIA_ROOT` naar een bekenede in-repo dir verwijst en schoonmaakt na het testen is verder prima).

Reply:
Aha, ik kan ook gewoon de content diect in de django file stoppen inplaats van in eerst de temp file. Door dit te gebruiken word de media root niet eens gevult waardoor de media override ook niet meer nodig is.

Reply:
Nog makkelijker idd.